nncf/pruning/model_analysis.py

```diff
@@ -0,0 +1,243 @@
+ limitations under the License.
+"""
+import networkx as nx
+
+from nncf.nncf_network import NNCFNetwork
+from nncf.pruning.export_helpers import PRUNING_OPERATOR_METATYPES, Convolution, \
+    Concat, StopMaskForwardOps
+from nncf.pruning.utils import is_depthwise_conv, find_next_nodes_not_types
+
+
+# pylint: disable=protected-access
+class NodesCluster:
+    def __init__(self, cluster_id, nodes, nodes_orders):
+        self.id = cluster_id
+        self.nodes = nodes
+        self.importance = max(nodes_orders)
+
+    def clean_cluster(self):
+        self.nodes = []
+        self.importance = 0
+
+    def add_nodes(self, nodes):
+        self.nodes.extend(nodes)
+
+
+class Clusterization:
+    def __init__(self, id_attr_name="id"):
+        self.clusters = {}
+        self._node_to_cluster = {}
+        self.id_attr = id_attr_name
+
+    def get_cluster_by_id(self, cluster_id):
+        if cluster_id in self.clusters:
+            return self.clusters[cluster_id]
+        raise IndexError('No cluster with such index')
+
+    def get_cluster_for_node(self, node_id):
+        return self.clusters[self._node_to_cluster[node_id]]
+
+    def is_not_in_any_cluster(self, node_id):
+        return node_id in self._node_to_cluster
+
+    def add_cluster(self, cluster: NodesCluster, cluster_id):
+        if cluster_id in self.clusters:
+            raise IndexError('Cluster with such index already exist')
+        self.clusters[cluster_id] = cluster
+        for node in cluster.nodes:
+            self._node_to_cluster[getattr(node, self.id_attr)] = cluster_id
+
+    def delete_cluster(self, cluster_id):
+        for node in self.clusters[cluster_id].nodes:
+            node_id = getattr(node, self.id_attr)
+            self._node_to_cluster.pop(node_id)
+        self.clusters.pop(cluster_id)
+
+    def get_all_clusters(self):
+        return self.clusters.values()
+
+    def get_all_nodes(self):
+        all_nodes = []
+        for cluster in self.clusters.values():
+            all_nodes.extend(cluster.nodes)
+        return all_nodes
+
+    def merge_clusters(self, first_id, second_id):
+        cluster_1 = self.clusters[first_id]
+        cluster_2 = self.clusters[second_id]
+        if cluster_1.importance > cluster_2.importance:
+            cluster_1.add_nodes(cluster_2.nodes)
+            for node in cluster_2.nodes:
+                self._node_to_cluster[getattr(node, self.id_attr)] = first_id
+            self.clusters.pop(second_id)
+        else:
+            cluster_2.add_nodes(cluster_1.nodes)
+            for node in cluster_1.nodes:
+                self._node_to_cluster[getattr(node, self.id_attr)] = second_id
+            self.clusters.pop(first_id)
+
+
+def get_position(nx_nodes_list, idx):
+    for i, node in enumerate(nx_nodes_list):
+        if node['id'] == idx:
+            return i
+    return None
+
+
+def unit_clusters_for_nodes(nodes_to_merge, clusterization):
```

Comment:
Need a description of what this function does. I guess you meant the word "unite" in the beginning. "merge" is better in this case.